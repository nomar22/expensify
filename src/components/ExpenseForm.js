import 'react-dates/initialize';
import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

//normal Javascript date
// const date = new Date();

const now = moment();




class ExpenseForm extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }


    onDescriptionChange = (e) => {
        const description = e.target.value;

        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;


        this.setState({ note });
    };
    onAmountChange = (e) => {
        const amount = e.target.value;

        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState({ amount });

        }
    }

    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    };

    onFocusChange = ({ focused }) => {

        this.setState(() => ({ calendarFocused: focused }));
    };

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.description || !this.state.amount) {
            //Set Error state equal to 'Please Provide description and amount'
            this.setState({ error: 'Please Provide description and amount' });

        } else {
            //clear The error
            this.setState({ error: '' });
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            });
        }
    };



    render() {

        return (
            <div>
                {this.state.error && <p style={{ color: 'red' }}>{this.state.error} </p>}
                <form onSubmit={this.onSubmit}>
                    <div className="input-group">
                        <div className="input-group__item">
                            <input className="text-input"
                                type="text"
                                placeholder="Description"
                                value={this.state.description}
                                onChange={this.onDescriptionChange}
                                autoFocus />
                        </div>
                        <div className="input-group__item">
                            <input
                                className="text-input"
                                type="text"
                                placeholder="Amount"
                                value={this.state.amount}
                                onChange={this.onAmountChange}
                            />
                        </div>
                        <div className="input-group__item">
                            <SingleDatePicker
                                date={this.state.createdAt}
                                onDateChange={this.onDateChange}
                                focused={this.state.calendarFocused}
                                numberOfMonths={1}
                                isOutsideRange={() => (false)}
                                onFocusChange={this.onFocusChange}
                            />

                        </div>
                        <div className="input-group__item">
                            <textarea placeholder="Add a note to your Expense (optional)"

                                value={this.state.note}
                                className="text-area"
                                onChange={this.onNoteChange} />
                        </div>
                        <div className="input-group__item">
                            <button>Add Expense </button>
                        </div>
                    </div>






                </form>
            </div>
        );
    }
}

export default ExpenseForm;