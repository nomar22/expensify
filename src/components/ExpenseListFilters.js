import 'react-dates/initialize';
import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, endDate, setEndDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates';
// import { start } from 'repl';

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);

    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    };

    onSortChange = (e) => {
        e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount();
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };
    render() {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange} />

                <select onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false}

                />
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
});


export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);