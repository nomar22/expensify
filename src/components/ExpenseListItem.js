import React from 'react';

// description, amount, createdAt
export default (props)=>(
    <div>
        This Expense Item
    </div>
);

