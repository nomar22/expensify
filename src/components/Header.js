import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';
import {connect } from 'react-redux';
import {startLogout} from '../actions/auth'




export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" > DashBoard /</NavLink>
        <NavLink to="/create" activeClassName="is-active" > Criar /</NavLink>
        <NavLink to="/help" activeClassName="is-active"> Ajuda</NavLink>
        <button onClick={startLogout}> Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch)=>({
    startLogout:()=>dispatch(startLogout())
});
export default connect(undefined, mapDispatchToProps)(Header);