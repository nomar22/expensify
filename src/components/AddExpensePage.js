import React from 'react';
import ExpenseForm from './expenseForm';
const AddExpensePage = () => (
    <div>
      <h1> Add Expense</h1>
      <ExpenseForm />
    </div>
);
export default AddExpensePage;