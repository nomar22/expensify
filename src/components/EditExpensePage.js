import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense, startEditExpense } from '../actions/expenses';
import Modal from 'react-modal';



export class EditExpensePage extends React.Component {

    constructor(props) {
        // Modal.setAppElement('#app');
        super(props);
        this.state = {
           modalIsOpen:false
        };
    }
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/dashboard');
    };
    onRemove = () => {
        this.setState({modalIsOpen:true});
    };
    onConfirm = ()=>{
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/dashboard');

    }
    closeModal = ()=>{
        this.setState({modalIsOpen:false});
    }


    render() {
        return (
            <div >
                <div className="page-header"  >
                    <div className="content-container">
                        <h1 className="page-header__title"> Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <div >
                        <button className="button--secondary" onClick={this.onRemove} >Remove Expense</button>
                    </div>

                </div>
                <Modal
                    isOpen={this.state.modalIsOpen}
                    className="modal"
                    onRequestClose={this.closeModal}
                >

                    <h2 ref={subtitle => this.subtitle = subtitle}>Delete this item?</h2>
                    <button className="button-layout" onClick={this.onConfirm}>OK</button>
                </Modal>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});
const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);