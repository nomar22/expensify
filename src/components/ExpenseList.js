import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem'

const ExpenseList =  (props)=>(

    <div> 
        <h1>ExpenseList</h1>
        Size {props.expenses.length  }
        
        <ol>
            {
                props.expenses.map((expense)=>(
                    <ExpenseListItem key={expense.id} {...expense}
                    />
                ))
            }
        </ol>
    </div>
);

const mapStateToProps = (state)=>{
    return {
        
        expenses:state.expenses,
        filters: state.filters
    };
};

export default  connect(mapStateToProps)(ExpenseList);

