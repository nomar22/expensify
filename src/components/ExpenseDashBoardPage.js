import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
export const ExpenseDashBoardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);


export default ExpenseDashBoardPage;