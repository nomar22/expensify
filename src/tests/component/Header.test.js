import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../components/Header';
import { wrap } from 'module';

// // 
let startLogout, wrapper;
beforeEach(()=>{
    startLogout = jest.fn();
    wrapper = shallow(<Header startLogout={startLogout} />);
});

test('Should render Header correctly', ()=>{
    expect(wrapper).toMatchSnapshot();
});


test('Should call startLogout on button click',()=>{
    wrapper.find('button').simulate('click');

    expect(startLogout).toHaveBeenCalled();


});

// should call startlogout
//  loginPage test file ->should call start login on button click

// spies