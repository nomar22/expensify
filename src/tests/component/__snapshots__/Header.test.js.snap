// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/dashboard"
  >
     DashBoard /
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
     Criar /
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
     Ajuda
  </NavLink>
  <button
    onClick={[Function]}
  >
     Logout
  </button>
</header>
`;
